test/jest/types.spec.ts

```diff
@@ -0,0 +1,593 @@
+            expect(getIdentityName(Google)).toBe('google');
+            expect(getIdentityName(Microsoft)).toBe('microsoft');
+            expect(getIdentityName(Yahoo)).toBe('yahoo');
+            expect(getIdentityName(Email)).toBe('email');
+            expect(getIdentityName(FacebookCustomAudienceId)).toBe(
+                'facebookcustomaudienceid'
+            );
+            expect(getIdentityName(Other2)).toBe('other2');
+            expect(getIdentityName(Other3)).toBe('other3');
+            expect(getIdentityName(Other4)).toBe('other4');
+            expect(getIdentityName(Other5)).toBe('other5');
+            expect(getIdentityName(Other6)).toBe('other6');
+            expect(getIdentityName(Other7)).toBe('other7');
+            expect(getIdentityName(Other8)).toBe('other8');
+            expect(getIdentityName(Other9)).toBe('other9');
+            expect(getIdentityName(Other10)).toBe('other10');
+            expect(getIdentityName(MobileNumber)).toBe('mobile_number');
+            expect(getIdentityName(PhoneNumber2)).toBe('phone_number_2');
+            expect(getIdentityName(PhoneNumber3)).toBe('phone_number_3');
+        });
+
+        it('returns null if the identity type is not found', () => {
+            const { getIdentityName } = IdentityType;
+
+            expect(getIdentityName('foo')).toBe(null);
+        });
+    });
+
+    describe('#getName', () => {
+        it('returns the name of an Identity Type', () => {
+            const { getName } = IdentityType;
+
+            expect(getName(CustomerId)).toBe('Customer ID');
+            expect(getName(Facebook)).toBe('Facebook ID');
+            expect(getName(Twitter)).toBe('Twitter ID');
+            expect(getName(Google)).toBe('Google ID');
+            expect(getName(Microsoft)).toBe('Microsoft ID');
+            expect(getName(Yahoo)).toBe('Yahoo ID');
+            expect(getName(Email)).toBe('Email');
+            expect(getName(FacebookCustomAudienceId)).toBe('Facebook App User ID');
+        }); 
+
+        it('returns other if the identity type is not found', () => {
+            const { getName } = IdentityType;
+
+            expect(getName(Other)).toBe('Other ID');
+            expect(getName(Other2)).toBe('Other ID');
+            expect(getName(Other3)).toBe('Other ID');
+            expect(getName(Other4)).toBe('Other ID');
+            expect(getName(Other5)).toBe('Other ID');
+            expect(getName(Other6)).toBe('Other ID');
+            expect(getName(Other7)).toBe('Other ID');
+            expect(getName(Other8)).toBe('Other ID');
+            expect(getName(Other9)).toBe('Other ID');
+            expect(getName(Other10)).toBe('Other ID');
+            expect(getName(MobileNumber)).toBe('Other ID');
+            expect(getName(PhoneNumber2)).toBe('Other ID');
+            expect(getName(PhoneNumber3)).toBe('Other ID');
+
+            expect(getName(NaN)).toBe('Other ID');
+        });
```

Comment:
I think you uncovered a bug here.  all these should probably have an actual Name that's not Other Id.  It's likely that the `getName` switch statement never was updated after these new IdentityTypes were added.

Reply:
Created https://go.mparticle.com/work/SQDSDKS-6942 to track